core/math/vector2.cpp

```diff
@@ -119,11 +119,10 @@ Vector2 Vector2::round() const {
 }
 
 Vector2 Vector2::rotated(real_t p_by) const {
-
-	Vector2 v;
-	v.set_rotation(angle() + p_by);
-	v *= length();
-	return v;
+	real_t sin = Math::sin(p_by), cos = Math::cos(p_by);
```

Comment:
I would spell the full names ``cosine`` and ``sine`` to make it clear that they are variables and not functions. IMO it's confusing to name the variables the same name as the corresponding functions. 

Another nitpick, I usually prefer each variable definition is on its own line just to make reading the function easier.

Reply:
Makes sense, I didn't think it was a problem because of the need to prefix math functions with `Math::`. But if I name them `sine` and `cosine` it won't line up as neatly. What about `sine` and `cosi`?

Own lines sounds like a good idea.

Reply:
Your right, the names won't conflict because of the namespace, but for human readability sine and cosine are better. I wouldn't worry about lining up the lines. IMO descriptive names are more important than lining things up. But that is a judgement call for you to make. Do what you think is best.

Reply:
I think `sin` and `cos` are still very human readable, but it's nice that `sine` and `cosi` are different from the function names, and one of them is the full word, so I think they're best.